### PR TITLE
Fix Navbar to conditionally render Admin link

### PR DESCRIPTION
**Pull request by ARodriguezHacks**
PR for Issue #749 Conditionally render "Admin" link in navbar
https://github.com/codeforpdx/recordexpungPDX/issues/749

### CODE REVIEW

File: src/frontend/src/components/Header/index.tsx [hunk 1]
```diff
@@ -7,10 +7,12 @@ import { connect } from 'react-redux';
 
 interface Props {
   isAuthenticated: boolean;
+  isAdmin: boolean; // Added 'isAdmin' for conditional rendering to work (line 35/36)
 }
 
 class Header extends React.Component<Props> {
   public render() {
+    console.log(this.props);
```
Comment (KentShikama) [inline src/frontend/src/components/Header/index.tsx]: This looks like left over debugging statements.

File: src/frontend/src/redux/system/reducer.ts [hunk 2]
```diff
@@ -1,22 +1,26 @@
...
-  loggedIn: false
+  loggedIn: false,
+  isAdmin: false
 };
 
 export function systemReducer(
   state = initialState,
   action: SystemActionTypes
 ): SystemState {
+  console.log('action', action);
```
Comment (KentShikama) [inline src/frontend/src/redux/system/reducer.ts]: This looks like left over debugging statements.

File: src/frontend/package.json [hunk 3]
```diff
@@ -20,13 +20,13 @@
...
     "moment": "^2.24.0",
-    "node-sass": "^4.13.0",
+    "node-sass": "^4.13.1",
     "prettier": "^1.19.1",
     "react": "^16.12.0",
     "react-dom": "^16.12.0",
     "react-redux": "^7.1.3",
     "react-router-dom": "^5.1.2",
-    "react-scripts": "3.2.0",
+    "react-scripts": "^3.3.0",
```
Comment (KentShikama) [inline src/frontend/package.json]: While these probably need to be updated, I would prefer if they are done in a separate PR. You can use 

```
git reset HEAD~1 -- src/frontend/package.json
git reset HEAD~1 -- src/frontend/package-lock.json
```

and commit that.

File: src/frontend/src/components/Header/index.tsx [hunk 4]
```diff
@@ -30,12 +32,15 @@ class Header extends React.Component<Props> {
             >
               Stats
             </button>
-            <button
-              onClick={() => history.push('/admin')}
-              className="link mid-gray hover-blue f6 f5-ns dib pa3"
-            >
-              Admin
-            </button>
+            {/* Conditionally render the admin button based on isAdmin from the props */}
```
Comment (KentShikama) [inline src/frontend/src/components/Header/index.tsx]: In general I prefer locally scoped comments like this one only if the code does something that is surprising giving the context or unusually unreadable.

File: src/frontend/src/components/Header/index.tsx [hunk 5]
```diff
@@ -62,7 +67,9 @@ class Header extends React.Component<Props> {
 }
 
 const mapStateToProps = (state: AppState) => ({
-  isAuthenticated: state.system.loggedIn
+  isAuthenticated: state.system.loggedIn,
+  isAdmin: state.system.isAdmin
+  //Do the same as above but with 'isAdmin'
```
Comment (KentShikama) [inline src/frontend/src/components/Header/index.tsx]: What does "above" refer to here?

File: src/frontend/src/redux/system/reducer.ts [hunk 6]
```diff
@@ -1,22 +1,26 @@
...
-        loggedIn: true
+        loggedIn: true,
+        isAdmin: action.isAdmin
       };
     }
     case LOG_OUT: {
       return {
-        loggedIn: false
+        loggedIn: false,
+        isAdmin: false
```
Comment (KentShikama) [inline src/frontend/src/redux/system/reducer.ts]: `isAdmin` should not exist for log out.

File: src/frontend/src/components/Header/index.tsx [hunk 7]
```diff
@@ -7,10 +7,12 @@ import { connect } from 'react-redux';
 
 interface Props {
   isAuthenticated: boolean;
+  isAdmin: boolean; // Added 'isAdmin' for conditional rendering to work (line 35/36)
```
Comment (KentShikama) [inline src/frontend/src/components/Header/index.tsx]: Comments like these become quickly outdated (especially line numbers). The commit message is probably a better place for this comment.

File: src/frontend/src/redux/system/reducer.ts (hunk 6 above)
Comment (ARodriguezHacks) [reply src/frontend/src/redux/system/reducer.ts]: I'm not sure how to resolve this issue. Tried to remove it bu received an error: "Property 'isAdmin' is missing in type '{ loggedIn: false; }' but required in type 'SystemState'."

Comment (KentShikama) [reply src/frontend/src/redux/system/reducer.ts]: https://github.com/codeforpdx/recordexpungPDX/pull/783/files#diff-076570d447b7ed9a8929b014215fae08R3 makes it required. You could change it to `isAdmin?: boolean;`. If for whatever reason that doesn't work, then keeping line 22 above is fine but we probably should add it to the types at https://github.com/codeforpdx/recordexpungPDX/pull/783/files#diff-076570d447b7ed9a8929b014215fae08R14 too.

Comment (wittejm) [reply src/frontend/src/redux/system/reducer.ts]: It looks to me like `isAdmin` can't be made optional here so long as it's a required field in SystemState (which is good) , so apart from Kent's alternate suggestion here this PR looks good.

Comment (ARodriguezHacks) [reply src/frontend/src/redux/system/reducer.ts]: If only I saw this comment three minutes ago. Oh well, I already changed it to reflect Kent's suggestion, and I checked that the log in and log out were still working properly.
